ipn/ipnlocal/local.go

```diff
@@ -1220,6 +1229,9 @@ func (b *LocalBackend) SetControlClientStatus(c controlclient.Client, st control
 		prefs.WantRunning = true
 		prefs.LoggedOut = false
 	}
+	if shouldAutoExitNode() && b.lastSuggestedExitNode == "" {
+		b.suggestExitNodeLocked()
```

Comment:
I think the netmap should be made into a parameter here, so it can compute based on `st.NetMap` (the updated one) instead of `b.netMap` (the previous netmap). Note that `st.NetMap` might be `nil`, in which case it needs to use `b.netMap`.